error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/pure-function-1.rs:20:22
   |
20 | fn fail (x: i32,) -> bool {
   |                      ^^^^
   |
note: the error originates here
  --> $DIR/pure-function-1.rs:20:1
   |
20 | / fn fail (x: i32,) -> bool {
21 | |     let y = foo(x);
22 | |     if bar(y) {
23 | |         false
...  |
26 | |     }
27 | | }
   | |_^
note: counterexample for "x"
       value:   -2
  --> $DIR/pure-function-1.rs:20:10
   |
20 | fn fail (x: i32,) -> bool {
   |          ^
note: counterexample for "y"
       value:   3
  --> $DIR/pure-function-1.rs:21:13
   |
21 |     let y = foo(x);
   |             ^^^^^^
note: counterexample for "result"
       value:   false
  --> $DIR/pure-function-1.rs:23:9
   |
23 |         false
   |         ^^^^^
note: counterexample for "foo()"
       value:   3
  --> $DIR/pure-function-1.rs:21:13
   |
21 |     let y = foo(x);
   |             ^^^^^^
note: counterexample for "bar()"
       value:   true
  --> $DIR/pure-function-1.rs:22:8
   |
22 |     if bar(y) {
   |        ^^^^^^

error: aborting due to previous error

